### ISSUE TITLE
is it possible to make shouldStopPolling run async code? 

### ISSUE THREAD

**Issue by DeanKamali** [kleinfreund/poll#5]
I have a use case where I need to run async code in `shouldStopPolling`; adding `async shouldStopPolling` breaks pooling functionally, and `shouldStopPolling` check runs only once.

**Comment by kleinfreund** [comment]
Hey,

that’s simple enough to add. I’ll look into it.

**Comment by DeanKamali** [comment]
Thank you, 
I'm not using Typescript in my project, so looking at `dist/poll.js` I made the following changes to make it work 
```
async function a(a, e, i = (() => !1)) {
    e = Math.max(0, e);
    do {
      if (await a(), await i()) break;
      await new Promise((a => setTimeout(a, e)))
    } while (!await i())
  }
```

**Comment by kleinfreund** [comment]
Yes, these are the changes I’m looking to implement. They’re trivial to implement itself, but I’m running into an unexpected issue with the tests that start to crash (see https://github.com/facebook/jest/issues/11713#issuecomment-957993206 for a description) so I can’t release them now. I will have to look into this on the weekend.

**Comment by kleinfreund** [comment]
@DeanKamali Hello again. I’ve added support for this feature in https://github.com/kleinfreund/poll/commit/29372356516b7b84a81ccb22caab6f9276d968a9. A new version of the package has just been published (see [release notes for v3.1.0](https://github.com/kleinfreund/poll/releases/tag/v3.1.0)).

**Comment by DeanKamali** [comment]
Thank you, I can confirm it works!
